docs/control-plane-configuration.md

```diff
@@ -0,0 +1,56 @@
+# Control Plane Configuration
+
+This document describes how to dynamically update the NGINX Kubernetes Gateway control plane configuration.
+
+## Overview
+
+NGINX Kubernetes Gateway offers a way to update the control plane configuration dynamically without the need for a
+restart. The control plane configuration is stored in the NginxGateway custom resource. This resource is created
+during the installation of NGINX Kubernetes Gateway.
+
+If using manifests, the default name of the resource is `nginx-gateway-config`. If using Helm, the default name
+of the resource is `<release-name>-config`. It is deployed in the same Namespace as the controller
+(default `nginx-gateway`).
+
+The control plane only watches this single instance of the custom resource. If the resource is invalid per the OpenAPI
+schema, the Kubernetes API server will reject the changes. If the resource is deleted or deemed invalid by NGINX
+Kubernetes Gateway, an error Event is created in the `nginx-gateway` Namespace, and the default values will be used by
```

Comment:
based on this https://github.com/nginxinc/nginx-kubernetes-gateway/blob/3a879ef24a5e93da02941e053b40abd732754547/internal/mode/static/handler.go#L133 , the event is a warning not error.